website_mrp_bom/views/website_mrp_bom_templates.xml

```diff
@@ -0,0 +1,90 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <template id="mrp_bom_page" name="MRP BoM Page">
+        <t t-call="website.layout">
+            <div id="wrap">
+                <div class="oe_structure"/>
+                <div class="container">
+                    <div class="row">
+                        <t t-call="website_mrp_bom.mrp_bom_detail_header"/>
+                        <t t-call="website_mrp_bom.mrp_bom_detail_table"/>
+                    </div>
+                </div>
+                <div class="oe_structure"/>
+            </div>
+        </t>
+    </template>
+
+    <template id="mrp_bom_detail_header" name="MRP BoM Details Header">
+        <h1 class="col-md-12 text-center" id="mrp_bom_name" t-field="bom.product_tmpl_id.name"/>
+        <div class="col-md-4">
+            <t t-if="bom.product_tmpl_id">
+                <div t-field="bom.product_tmpl_id.image" t-options='{"widget": "image", "class": "center-block mb16", "max_width": 128}'/>
+            </t>
+            Base Quantity:
+            <div t-field="bom.product_qty"/>
+            <div groups="product.group_uom" t-field="bom.product_uom_id.name"/>
+        </div>
+        <div class="col-md-8 mt32">
+            <t t-if="bom">
+                <div t-field="bom.website_description"/>
+            </t>
+        </div>
+    </template>
+
+    <template id="mrp_bom_detail_table" name="MRP BoM Details Table">
+        <div class="col-md-12">
+            <table class="table">
+                <thead>
+                    <tr>
+                        <th>Product</th>
+                        <th/>
+                        <th>Quantity</th>
+                        <t t-set="has_bom" t-value="False"/>
+                        <t t-foreach="bom.bom_line_ids" t-as="bom_line">
+                            <t t-if="bom_line.product_id.sudo().bom_ids and bom_line.product_id.sudo().bom_ids[0].website_published">
+                                <t t-set="has_bom" t-value="True"/>
+                            </t>
+                        </t>
+                        <t t-if="has_bom">
+                            <th>BoM</th>
+                        </t>
+                    </tr>
+                </thead>
+                <tbody>
+                    <t t-foreach="bom.bom_line_ids" t-as="line">
+                        <t t-set="bom_product_id" t-value="line.product_id.sudo()"/>
```

Comment:
isn't the product published automatically?

Reply:
No. If it not published, on components list it will appear without href link. When published you can access to a description using website_product module.